compliance/collection/nodeinventorizer/caching_inventorizer.go

```diff
@@ -101,16 +101,19 @@ func min(d1 time.Duration, d2 time.Duration) time.Duration {
 
 func readBackoff(path string) time.Duration {
 	wrap := readInventoryWrap(path)
-	if wrap != nil {
-		return wrap.RetryBackoffDuration
+	if wrap == nil {
+		return 0
+	}
+	if d, err := time.ParseDuration(wrap.RetryBackoffDuration); err == nil {
+		return d
 	}
 	return 0
 }
```

Comment:
Sorry, did not spot this before.

Feels like we should change the defaulting logic in case reading or parsing fails. If it fails, we don't know whether we should apply a backoff or not. The more conservative choice is to apply the max backoff. This will prevent us overloading the node in case the previous scans failed but the written file is corrupt.

The problem currently is that `readInventoryWrap` does not differentiate between absent wrap (absent backoff) and error reading one. I feel at this point it would make sense to have a deeper change.
One option is to make `readInventoryWrap` return `error` and handle that differently in `readBackoff` and `readCachedInventory`. There are other options though.

Reply:
Great catch!
I added an additional parameter (`errorBackoff`) to the function to keep it independent from the struct like the rest of the helper functions. Please TAL at [176533b](https://github.com/stackrox/stackrox/pull/4701/commits/176533b922cd57c9288ea36df1b3bf475c7d21ab) for the change and its according test 😃